packages/vega-parser/src/parsers/guides/axis-grid.js

```diff
@@ -37,26 +39,100 @@ export default function(spec, config, userEncode, dataRef, band) {
     round:  _('tickRound')
   };
 
-  if (orient === Top || orient === Bottom) {
-    u = 'x';
-    v = 'y';
-    s = 'height';
-  } else {
-    u = 'y';
-    v = 'x';
-    s = 'width';
-  }
-  v2 = v + '2';
+  if (isSignal(orient)) {
+    val = {}
+    for (var key of Object.keys(tickPos)) {
+      val[key] = topOrBottomAxisExpr(orient.signal, tickPos[key],)
+    }
+
+    update['x'] = enter['x'] = {
+      scale: topOrBottomAxisExpr(orient.signal, tickPos.scale, vscale ? vscale : undefined),
```

Comment:
Can vscale be false or null or what's this check for?

Reply:
looks like it, this is following the original logic for non-signal orient (in this file)